src/components/transport_manager/CMakeLists.txt

```diff
@@ -49,79 +49,75 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "Linux")
 endif ()
 
-set(LIBRARIES
-  ProtocolLibrary
-  Utils
+set(PATHS
+  ${CMAKE_CURRENT_SOURCE_DIR}/include
+  ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
 
-set(SOURCES
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_manager_default.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_manager_impl.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/network_interface_listener_impl.cc
-
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_client_listener.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_connection_factory.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_device.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_server_originated_socket_connection.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_socket_connection.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_transport_adapter.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/tcp_transport_adapter.cc
-
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_adapter/threaded_socket_connection.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_adapter/transport_adapter_impl.cc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/transport_adapter/transport_adapter_listener_impl.cc
-
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/iap2_emulation/iap2_transport_adapter.cc
+set(EXCLUDE_PATHS)
 
+set(LIBRARIES
+  ProtocolLibrary
+  Utils
 )
 
-if (CMAKE_SYSTEM_NAME STREQUAL "QNX")
-  list(APPEND SOURCES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/platform_specific/qnx/platform_specific_network_interface_listener.cc
-  )
-elseif (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-  list(APPEND SOURCES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/tcp/platform_specific/linux/platform_specific_network_interface_listener.cc
-  )
-endif ()
-
 if(BUILD_BT_SUPPORT)
   list(APPEND LIBRARIES
     bluetooth
   )
-  list(APPEND SOURCES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_connection_factory.cc
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_device.cc
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_device_scanner.cc
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_socket_connection.cc
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth/bluetooth_transport_adapter.cc
+else()
+  list(APPEND EXCLUDE_PATHS
+    ${COMPONENTS_DIR}/transport_manager/include/transport_manager/bluetooth
+    ${COMPONENTS_DIR}/transport_manager/src/bluetooth
   )
 endif()
 
 if(BUILD_USB_SUPPORT)
-  list(APPEND SOURCES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/usb_aoa_adapter.cc
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/usb_connection_factory.cc
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/usb_device_scanner.cc
-  )
   if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    list(APPEND SOURCES
-      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/libusb/platform_usb_device.cc
-      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/libusb/usb_connection.cc
-      ${CMAKE_CURRENT_SOURCE_DIR}/src/usb/libusb/usb_handler.cc
+    set(EXCLUDE_PATHS
+      ${COMPONENTS_DIR}/transport_manager/include/transport_manager/usb/qnx
+      ${COMPONENTS_DIR}/transport_manager/src/usb/qnx
     )
+
   elseif(CMAKE_SYSTEM_NAME STREQUAL "QNX")
+    set(EXCLUDE_PATHS
```

Comment:
Better to use `list(APPEND` here rather than set, otherwise you might overwrite the variable if someone sets this value above this line in the future

Reply:
Never mind, this is an existing bug, does not need to be fixed here